sessionctx/stmtctx/stmtctx.go

```diff
@@ -173,6 +172,8 @@ type StatementContext struct {
 	// Map to store all CTE storages of current SQL.
 	// Will clean up at the end of the execution.
 	CTEStorageMap interface{}
+	// cachedTables is used to store cache table id when it satisfies the cache read condition
+	cachedTables map[int64]bool
```

Comment:
Maybe a slice is better here?
```
cachedTables []struct{
    id int64
    used bool
}
```

There won't be much cached table here, so the performance of slice should be good enough.
And this is in the performance critical path, reuse slice to avoid allocation is much easier, just `cachedTables = cachedTables[:0]`

Reply:
okay